src/banyan_native/operations/prepare.rs

```diff
@@ -9,11 +16,106 @@ use std::{
+///
+/// * `fs` - FileSystem to modify
+/// * `omni` - Context aware online / offline Drive
+/// * `client` - Means of connecting to the server if need be
+/// * `follow_links` - Whether or not to follow symlinks when bundling.
+///
+/// # Return Type
+/// Returns `Ok(())` on success, otherwise returns an error.
+pub async fn pipeline(
+    mut fs: FsMetadata,
+    omni: &mut OmniBucket,
+    client: &mut Client,
+    follow_links: bool,
+) -> Result<String, TombError> {
+    // Local is non-optional
+    let mut local = omni.get_local()?;
+
+    // If there is a remote Bucket with metadatas that include a content root cid which has already been persisted
+    if client.is_authenticated().await {
+        if let Ok(remote) = omni.get_remote() {
+            if let Ok(metadatas) = Metadata::read_all(remote.id, client).await {
+                if metadatas.iter().any(|metadata| {
+                    Some(metadata.root_cid.clone())
+                        == local.content.get_root().map(|cid| cid.to_string())
+                }) {
+                    info!("Starting a new delta...");
+                    local.content.add_delta()?;
+                    omni.set_local(local.clone());
+                }
+            }
+        }
+    }
 
-use super::wnfsio::get_progress_bar;
+    // Create bundling plan
+    let bundling_plan = create_plans(&local.origin, follow_links).await?;
+
+    // Get all the paths present on disk
+    let mut all_disk_paths = <Vec<PathBuf>>::new();
+    for plan in bundling_plan.clone() {
+        match plan {
+            PreparePipelinePlan::Directory(metadata)
+            | PreparePipelinePlan::Symlink(metadata, _) => {
+                all_disk_paths.push(metadata.original_location.clone());
+            }
+            PreparePipelinePlan::FileGroup(metadatas) => {
+                let paths: Vec<PathBuf> = metadatas
+                    .iter()
+                    .map(|metadata| metadata.original_location.clone())
+                    .collect();
+                all_disk_paths.extend(paths);
+            }
+        }
+    }
+
+    // Get all nodes and their associated paths
+    let all_node_paths = fs.get_all_nodes(&local.metadata).await?;
+
+    // Track all blocks removed since the last preparation
+    for (node, wnfs_path) in all_node_paths {
+        // If the existing WNFS node is not still represented on disk
+        if !all_disk_paths.contains(&wnfs_path) {
+            // If the node is a File, add all the CIDs associated with it to a list
+            if let PrivateNode::File(file) = node {
+                local
+                    .deleted_block_cids
+                    .extend(file.get_cids(&fs.forest, &local.metadata).await?);
+            }
+            // Remove the reference from the WNFS
+            fs.rm(&path_to_segments(&wnfs_path)?, &local.metadata)
+                .await?;
+        }
+    }
+
+    let split_store_local = DoubleSplitStore::new(&local.content, &local.metadata);
```

Comment:
( this is from previous PR; code moved )
yes because this is preparation / encryption. might need a refactor but is big cucumber not part of this pra